app/controllers/users_controller.rb

```diff
@@ -0,0 +1,53 @@
+class UsersController < ApplicationController
+  def index
+    if session[:user_id]
+      @users = User.all
+    else
+      flash.now[:error] = "You must be logged in to see this page"
```

Comment:
Usually you would redirect but this is an interesting solution that seems like it will work just as well.  

Good job.  😃